core/src/main/java/org/elasticsearch/action/search/SearchPhaseExecutionException.java

```diff
@@ -144,15 +144,20 @@ protected void innerToXContent(XContentBuilder builder, Params params) throws IO
             builder.endObject();
         }
         builder.endArray();
-        super.innerToXContent(builder, params);
     }
 
     @Override
-    protected void causeToXContent(XContentBuilder builder, Params params) throws IOException {
-        if (super.getCause() != null) {
-            // if the cause is null we inject a guessed root cause that will then be rendered twice so wi disable it manually
-            super.causeToXContent(builder, params);
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        Throwable ex = ExceptionsHelper.unwrapCause(this);
+        if (ex != this) {
+            generateThrowableXContent(builder, params, this);
+        } else {
+            // We don't have a cause when all shards failed, but we do have shards failures so we can "guess" a cause.
+            // Here, we use super.getCause() because we don't want the guessed exception to be rendered twice (one in
+            // the "cause" field, one in "failed_shards")
```

Comment:
thanks for changing the explanation! makes more sense now. Can you also point it to the getCause method where we guess the cause?